controlplane/rosa/controllers/rosacontrolplane_controller.go

```diff
@@ -327,6 +328,35 @@ func (r *ROSAControlPlaneReconciler) reconcileDelete(ctx context.Context, rosaSc
 	}
 
 	if cluster.Status().State() != cmv1.ClusterStateUninstalling {
+		machinePools, err := rosaScope.GetMachinePools()
+		if err != nil {
+			rosaScope.Info("Error getting controlPlan machinePools ", err)
+			return ctrl.Result{}, err
+		}
+
+		var errs []error
+		for id, mp := range machinePools {
+			if mp.DeletionTimestamp != nil && !mp.DeletionTimestamp.IsZero() {
```

Comment:
```suggestion
			if !mp.DeletionTimestamp.IsZero() {
```
no need to check for nil, `IsZero()` already does that.